src/Hive.Core/sia380/sia380.py

```diff
@@ -269,58 +278,131 @@ def main(room_properties, floor_area, T_e, setpoints_ub, setpoints_lb, surface_a
+        gamma_lb_cooling_no_hr = (Q_i[month] + Q_s[month]) / (Q_T_ub + Q_V_lb_cooling_no_heat_recovery)
 
         # usage of heat gains (Ausnutzungsgrad für Wärmegewinne), eta_g
-        eta_g_lb = calc_eta_g(Q_T_lb, Q_V_lb, gamma_lb, tau)
-        eta_g_ub = calc_eta_g(Q_T_ub, Q_V_ub, gamma_ub, tau)
+        eta_g_lb_heating = calc_eta_g(gamma_lb_heating, tau, False)
+        eta_g_ub_heating = calc_eta_g(gamma_ub_heating, tau, False)
+        eta_g_lb_cooling = calc_eta_g(gamma_lb_cooling, tau, True)
+        eta_g_ub_cooling = calc_eta_g(gamma_ub_cooling, tau, True)
+        eta_g_lb_heating_no_hr = calc_eta_g(gamma_lb_heating_no_hr, tau, False)
+        eta_g_ub_heating_no_hr = calc_eta_g(gamma_ub_heating_no_hr, tau, False)
+        eta_g_lb_cooling_no_hr = calc_eta_g(gamma_lb_cooling_no_hr, tau, True)
+        eta_g_ub_cooling_no_hr = calc_eta_g(gamma_ub_cooling_no_hr, tau, True)
 
         # heating demand (Heizwärmebedarf), Q_H
-        demand_ub = Q_T_ub + Q_V_ub - eta_g_ub * (Q_i[month] + Q_s[month])
-        demand_lb = Q_T_lb + Q_V_lb - eta_g_lb * (Q_i[month] + Q_s[month])
-        eta_g = 0.0
-        if abs(demand_lb) < abs(demand_ub):
-            Q_T[month] = Q_T_lb
-            QT_opaque[month] = sum(QT_op_per_srf_this_month_lb)
-            QT_transparent[month] = sum(QT_tr_per_srf_this_month_lb)
-            Q_V[month] = Q_V_lb
-            eta_g = eta_g_lb
-        else:
-            Q_T[month] = Q_T_ub
-            QT_opaque[month] = sum(QT_op_per_srf_this_month_ub)
-            QT_transparent[month] = sum(QT_tr_per_srf_this_month_ub)
-            Q_V[month] = Q_V_ub
-            eta_g = eta_g_ub
-
-        Q_i_eta_g[month] = Q_i[month] * eta_g
-        Q_s_eta_g[month] = Q_s[month] * eta_g
-        demand = Q_T[month] + Q_V[month] - (Q_i_eta_g[month] + Q_s_eta_g[month])
-        if demand > 0:
-            Q_Heat[month] = demand
+        Q_H_ub = Q_T_ub + Q_V_ub_heating - eta_g_ub_heating * (Q_i[month] + Q_s[month])
+        Q_H_lb = Q_T_lb + Q_V_lb_heating - eta_g_lb_heating * (Q_i[month] + Q_s[month])
+        Q_H_ub_no_hr = Q_T_ub + Q_V_ub_heating_no_heat_recovery - eta_g_ub_heating_no_hr * (Q_i[month] + Q_s[month])
+        Q_H_lb_no_hr = Q_T_lb + Q_V_lb_heating_no_heat_recovery - eta_g_lb_heating_no_hr * (Q_i[month] + Q_s[month])
+        if Q_H_ub < 0:
+            Q_H_ub = 0
+        if Q_H_lb < 0:
+            Q_H_lb = 0
+        if Q_H_ub_no_hr < 0:
+            Q_H_ub_no_hr = 0
+        if Q_H_lb_no_hr < 0:
+            Q_H_lb_no_hr = 0
+
+        # cooling demand (Kältebedarf), Q_K
+        Q_K_lb = Q_i[month] + Q_s[month] - eta_g_lb_cooling * (Q_T_lb + Q_V_lb_cooling)
+        Q_K_ub = Q_i[month] + Q_s[month] - eta_g_ub_cooling * (Q_T_ub + Q_V_ub_cooling)
+        Q_K_lb_no_hr = Q_i[month] + Q_s[month] - eta_g_lb_cooling_no_hr * (Q_T_lb + Q_V_lb_cooling_no_heat_recovery)
+        Q_K_ub_no_hr = Q_i[month] + Q_s[month] - eta_g_ub_cooling_no_hr * (Q_T_ub + Q_V_ub_cooling_no_heat_recovery)
+        if Q_K_lb < 0:
```

Comment:
you could add a comment to this section, or even better, create a function:
`Q_K_lb = negative_to_zero(Q_K_lb)`

AND: make sure to use `0.0` anywhere you're using real numbers. Because `0` is an integer. You don't want that. Even though it can often be figured out correctly. Just use real numbers.

Reply:
I thought that is only for old python? but with `from __future__ import division` this won't be a problem anymore? Fixing it nevertheless